### ISSUE TITLE
being implemented into `docker compose`!

### ISSUE THREAD

**Issue by branchvincent** [ufoscout/docker-compose-wait#62]
👋 Thanks so much for this script! Been using it for a while and it's great

Just wanted to notify other users that this functionality is about to be merged into `docker compose`! See https://github.com/docker/compose/issues/8351 and https://github.com/docker/compose/pull/8777 if you want to leave any comments before its merged

**Comment by ufoscout** [comment]
@branchvincent Great to know it!
Anyway, this tool works not only in `docker compose`  but in any environment, for example in Kubernetes or in simple command line scripts, etc.

**Comment by briceburg** [comment]
I was thinking about this issue earlier (when it was still open) and wanted to comment.

the behavior of this tool is not being implemented into compose. the `--wait` flag being added to compose is intended to delay `docker-compose up` until all services with a healthcheck respond healthy. this greatly helps in CI/automation. It does not help with delaying a particular service's start until a condition is met (so that it can become healthy!) -- which is the purpose of this tool. for instance having a springboot service wait until a database is accepting connections before starting. if they start at same time there's a potential the springboot service crashes when initializing the DB as it may not be ready yet.  

this tool provides a portable means to control "wait" dependencies via environment variables and is leveraged by container entrypoints.

**Comment by ulgens** [comment]
I agree with @briceburg on the functionality on that PR and the one this repo provides are different things.
